### ISSUE TITLE
Question: there are some inconsistencies about label selectors

### ISSUE THREAD

**Issue by juangon** [quarkiverse/quarkus-operator-sdk#348]
Hi!
As what I understand, label selectors should be available in all resources in orther to be processed by reconciler. But in this case: https://github.com/java-operator-sdk/java-operator-sdk/blob/main/sample-operators/webpage/src/main/java/io/javaoperatorsdk/operator/sample/DeploymentDependentResource.java#L29 , label selector is different in the DependentResource class comparing to the Standalone Reconciler. How is this possible? Thanks very much

**Comment by csviri** [comment]
@juangon good point, that is a mistake. It works because those selectors are not mutually exclusive. Created a PR to fix this:
https://github.com/java-operator-sdk/java-operator-sdk/pull/1231

**Comment by csviri** [comment]
Pls create such issues in the core project. Not in the extension. 
But thank you for pointing this out!!

**Comment by juangon** [comment]
Thanks @csviri ! BTW, is it possible to have standalone dependent resources without having to add a label and label selector? I don't want users to explicitely add this label (I know I can create it dynamically for secondary resources but in the case of the primary resource I don't see any way for doing it).

**Comment by csviri** [comment]
@juangon any label is optional in all cases, but you can configure it also for the controller. You can access also all the configurations through `Context` object in `reconcile`.
